docs/rfc/rfc-103-incoming-txs-when-catching-up.md

```diff
@@ -0,0 +1,177 @@
+can dictate what honest nodes must send to their peers.
+
+The gossip algorithm currently implemented is a naive *push* protocol: a node
+will forward all the transactions in its mempool to all its peers, except to the
+nodes from which it received the transaction.
+
+The protocol implements only one minor optimization. For each peer, the
+`broadcastTxRoutine` [function](#broadcast) in the mempool reactor iterates
+through all transactions in the mempool and sends them one by one to a peer.
+Just before sending a transaction, if the node suspects that the peer is lagging
+behind, it waits some time before sending it. This is the [code](#optimization):
+```golang
+if peerState.GetHeight() < memTx.Height()-1 {
+  time.Sleep(PeerCatchupSleepIntervalMS * time.Millisecond)
+  continue
+}
+```
+where: 
+- `peerState` is the local state of the peer (updated through voting messages
+  received by the consensus reactor; see `PeerState.ApplyNewRoundStepMessage`), 
+- `memTx.Height()` is the height at which transaction `memTx.tx` was validated
+  (set during the handling of `CheckTx` responses), and
+- `PeerCatchupSleepIntervalMS` is fixed to `100`.
+
+>For a historical reference, this optimization has been around since 2015. There
+>is no documentation or comment on the reason it was introduced. A first version
+>with only the condition to check the peer's height was introduced in a
+>[commit](https://github.com/tendermint/tendermint/commit/12566f51af2bbdc73e3c79c603be0593d8cb1574)
+>from September 2015. In December 2015 a huge refactoring
+>[commit](https://github.com/CometBFT/cometbft/commit/ef43af19ab2af994afaf9fdb148df2918454d9c4)
+>introduced the line that makes the broadcast routine sleep.
+
+> Note that the evidence reactor implements a similar optimization.
+
+Should this code be removed from the mempool reactor to make it consistent with
+how the [RPC endpoints](#rpc) are implemented? Experimental evidence shows that
+actually the optimization works and improves the behavior of the nodes. The
+following picture shows the results of an experiment with four interconnected
+nodes. On the left we see the collected metrics when we run the nodes without
+the optimization. On the right we see the results of running the nodes with the
+optimization, that is, without modifying the code.
+![rfc-103-comparison](./images/rfc-103-optimization-comparison.png) The node in
+orange called _validator04_ joins the network at arount height 100 and starts
+performing BlockSync. In the graph at the bottom we can see the height of all
+nodes and in particular how this node starts from height 0 and catches up with
+the other nodes. Also we can observe that, when the optimization is disabled
+(left side), while the orange node is catching up, both its mempool size (top
+graph) and the number of rejected transactions (middle graph) increases
+significantly compared to the optimizated code (right side).
+
```

Comment:
Could you add a sentence to state the obvious :-)
Something like "In conclusion, the results presented above show that the optimization is indeed improving the system's performance and should be kept for the time being" (feel free to heavily reword this).